subsys/caf/modules/sensor_sampler.c

```diff
@@ -73,6 +96,104 @@ static size_t get_sensor_data_cnt(const struct sensor_config *sc)
 	return data_cnt;
 }
 
+static bool is_active(const struct sensor_config *sc, double curr, double prev)
+{
+	enum act_type type = sc->trigger->activation.type;
+	double tresh = sc->trigger->activation.tresh;
+
+	bool is_active = false;
+
+	switch (type) {
+	case ACT_TYPE_PERC:
+		is_active = fabs(curr - prev) > fabs(prev * tresh/100.0);
+		break;
+	case ACT_TYPE_ABS:
+		is_active = fabs(curr - prev) > fabs(tresh);
+		break;
+	default:
+		report_error();
+		break;
+	}
+
+	return is_active;
+}
+
+static bool can_sensor_sleep(const struct sensor_config *sc,
+			     struct sensor_data *sd,
+			     struct sensor_value *new)
+{
+	size_t data_cnt = get_sensor_data_cnt(sc);
+	bool sleep = true;
+
+	for (size_t i = 0; i < data_cnt; i++) {
+		double prev = sensor_value_to_double(&sd->prev[i]);
```

Comment:
You may consider keeping the old value as float (sensor_event uses floats to broadcast samples)

Reply:
I reworked the PR as suggested.